### DISCUSSION TITLE
Why alias Constraints to Assert?

### DISCUSSION THREAD

**Discussion by nsetyo** [symfony/symfony#45525] [Q&A] [answered]
The convention in Symfony aliasing `Symfony\Component\Validator\Constraints` to `Symfony\Component\Validator\Constraints as Assert`, why this is the default and not using `Constraints` directly? It has less text to type.

**Comment by mbabker** [comment, marked as answer]
It provides additional context to what you're doing, especially when you're in a class that might be using annotations/attributes from multiple packages.  Suppose you have a serializable User entity with validation constraints, it wouldn't be all that farfetched to have a property definition like this one:

```php
class User
{
    #[ORM\Column]
    #[Assert\Email]
    #[Serializer\Groups(['list'])]
    #[Gedmo\Versioned]
    public string $email;
}
```

With the aliased imports, you know where each of the attributes is coming from and what their purpose is (ORM field configuration, validation constraint, serialization group configuration, and the field is versioned with a Doctrine extension).  If you take away the imports, there's a bit more ambiguity with each attribute:

```php
class User
{
    #[Column]
    #[Email]
    #[Groups(['list'])]
    #[Versioned]
    public string $email;
}
```

Sure, you can figure it out in the context of your IDE by following the imports, but it adds a bit of mental overhead when reading the code.

**Reply by nsetyo** [reply]
Thanks for the explanation
